conans/client/cmd/uploader.py

```diff
@@ -295,8 +295,8 @@ def _upload_recipe(self, ref, conanfile, retry, retry_wait, policy, remote, remo
         if policy != UPLOAD_POLICY_FORCE:
             # Check SCM data for auto fields
             if hasattr(conanfile, "scm") and (
-                    conanfile.scm.get("url") == "auto" or conanfile.scm.get("revision") == "auto"):
-                raise ConanException("The recipe has 'scm.url' or 'scm.revision' with 'auto' "
+                    conanfile.scm.get("url") == None or conanfile.scm.get("url") == "auto" or conanfile.scm.get("revision") == "auto"):
```

Comment:
Having a second thought, if there are users that are doing ``url=None``, and now we start to block them: will they be surprised? Why are they doing this? The problem is that such information is not reproducible for conan builds (but the revision is there, they might use it in CI? Maybe with ``conan inspect`` is possible to read the ``revision`` and the URL is hardcoded in the CI?

Let's make sure we are not breaking other users without a clear value/alternative or without declaring it a bug.
Asking for @jgsogo feedback.

Reply:
when the url=None, it is not possible to upload anyway.

Reply:
If we want to cover in this PR the use-case where the SCM won't work because `url` is `None`, then we should cover other scenarios too: IMHO, if `type` is `None` or `revision` is `None` we should probably fail to upload the recipe.